### PR TITLE
implement cypress tests for flow-playground frontend

### LINKED ISSUE #335: Playground E2E Tests
Implement Cypress Test Suite for playground

Test cases:
- deploying contract to account
- executing script template
- executing transaction template
- updating storage of account
-

### PR DESCRIPTION
**Pull request by MrDSGC**
Closes: #335 

## Description

created cypress tests for 
- deploying contract scenario
- sending transaction scenario
- executing script scenario

### CODE REVIEW

File: cypress/e2e/playground.cy.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        cy.react('MenuList', {props: {title: 'Transaction Templates'}}).children().first().contains('Transaction')
+        cy.get('div[title=Transaction]').click()
+        cy.react('EditorTitle', {props: {type: 2}}).contains('Transaction Template')
+        // Ensure action button is disabled when contract not deployed yet
+        cy.react('ActionButton', {props: {type: 2, active: false}}).contains('Send')
+
+        // deploy contract
+        cy.react('AccountList').children().last().children().first().click()
+        cy.react('ActionButton', {props: {type: 1}}).click()
+        cy.wait(5000)
```
Comment (alse) [inline cypress/e2e/playground.cy.ts]: Waiting can be avoided by making an assertion on the page content. Cypress will automatically wait for that assertion and fail if it doesn't pass

File: cypress/e2e/playground.cy.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        cy.react('ActionButton', {props: {type: 1}}).contains('Redeploy')
+        cy.react('RenderResponse', {props: {resultType: 'CONTRACT'}}).should('exist')
+    })
+
+    it('sends a transaction', () =>{
+        cy.react('MenuList', {props: {title: 'Transaction Templates'}}).children().first().contains('Transaction')
+        cy.get('div[title=Transaction]').click()
+        cy.react('EditorTitle', {props: {type: 2}}).contains('Transaction Template')
+        // Ensure action button is disabled when contract not deployed yet
+        cy.react('ActionButton', {props: {type: 2, active: false}}).contains('Send')
```
Comment (alse) [inline cypress/e2e/playground.cy.ts]: cypress-react-selector looks pretty cool, but ideally we should test the DOM in these e2e tests. For example, in this case the disabled attribute never gets set on the button, but instead an empty function gets called. 

Using data attributes is also a nice way to keep tests valid even as component structure changes. (https://docs.cypress.io/guides/references/best-practices#Selecting-Elements)

We could look at some helper methods or page objects, but just using data-test attributes would be a great start.  Here's an example:

```
    cy.get('[data-test="action-button"]').click();

    cy.get('[data-test="action-button"]').should('be.disabled');
    cy.get('[data-test="control-panel-status-message"]').should(
      'have.text',
      'Please, wait...',
    );

    cy.get('[data-test="action-button"]').should('be.enabled');
    cy.get('[data-test="control-panel-status-message"]').should(
      'not.have.text',
      'Please, wait...',
    );
```

File: cypress/e2e/playground.cy.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      .get('iframe[id="FCL_IFRAME"]')
+      .its('0.contentDocument.body').should('not.be.empty')
+      .then(cy.wrap)
+  }
+  
+describe('Flow-Playground frontend tests', () => {
+
+    beforeEach(() => {
+        cy.viewport('macbook-16')
+        cy.visit('http://localhost:3000/')
```
Comment (alse) [inline cypress/e2e/playground.cy.ts]: Can we use a [baseUrl](https://docs.cypress.io/guides/references/best-practices#Setting-a-global-baseUrl) option?

File: cypress/e2e/playground.cy.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
+const getIframeBody = () => {
```
Comment (alse) [inline cypress/e2e/playground.cy.ts]: Is this used anywhere?

File: cypress/support/e2e.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
```
Comment (alse) [inline cypress/support/e2e.ts:18]: I think we should add `cypress/videos` to gitignore and remove the cypress/videos/playground.cy.ts.mp4 file

File: .github/workflows/integration-local.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+name: Flow-Playground Integration Tests
```
Comment (alse) [inline .github/workflows/integration-local.yml]: Could also reuse the existing ci.yml file

File: cypress.config.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { defineConfig } from "cypress";
+
+export default defineConfig({
+  projectId: "h1jieb",
+  video: true,
```
Comment (alse) [inline cypress.config.ts]: Should default video to false?

**Review by alse — COMMENTED** [review]
Nice @MrDSGC , great to have these tests! I left a few suggestions, let me know if you have any questions

File: cypress/e2e/playground.cy.ts (hunk 4 above)
Comment (MrDSGC) [reply cypress/e2e/playground.cy.ts]: nope it was an artifact from the tests i was referencing

File: cypress.config.ts (hunk 7 above)
Comment (MrDSGC) [reply cypress.config.ts]: is there a way to get video in the dashboard if defaulted to false?

Comment (alse) [reply cypress.config.ts]: Ah if we're planning on uploading the videos we could keep this option false and add a --record cli option when running the tests in the ci

File: src/components/RenderResponse/Line/index.tsx [hunk 8]
```diff
@@ -99,7 +99,7 @@ export const Line: React.FC<LineType> = ({ timestamp, tag, value, label }) => {
       {typeof value === 'string' ? (
         <StringValue tag={tag}>{value}</StringValue>
       ) : (
-        <ObjectValue>{JSON.stringify(value, null, 2)}</ObjectValue>
+        <ObjectValue data-test={dataTest} >{JSON.stringify(value, null, 2)}</ObjectValue>
```
Comment (alse) [inline src/components/RenderResponse/Line/index.tsx]: This dataTest value is duplicated in `src/components/RenderResponse/index.tsx`. Can it be something like `data-test="response-line"` instead?

File: cypress/e2e/playground.cy.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
+describe('Flow-Playground frontend tests', () => {
```
Comment (alse) [inline cypress/e2e/playground.cy.ts:1]: Could you add the tests directory to the prettier scripts in package.json

File: cypress/e2e/playground.cy.ts [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+        cy.get(STATUS_MESSAGE).should('have.text', 'Please, wait...');
+        cy.get(DEPLOY_BUTTON).should('have.text', 'Redeploy');
+        cy.get('[data-test="CONTRACT-response"]').should('include.text', 'Deployed Contract To: 0x01');
+    })
+
+    it('sends a transaction', () =>{
+        cy.get('[data-test="sidebar-Transaction"]').click();
+        cy.get('[data-test="editor-heading"]').should('include.text', 'Transaction Template');
+        // Ensure action button is disabled when contract not deployed yet
+        cy.get(SEND_BUTTON).should('have.text', 'Send');
```
Comment (alse) [inline cypress/e2e/playground.cy.ts]: Should this be something like `cy.get(SEND_BUTTON).should('be.disabled');`?

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -26,3 +33,33 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
       - run: make ci
+
+  cypress-e2e:
+    name: Cypress Emulator Tests for Flow-Playground
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    container: cypress/browsers:node16.14.2-slim-chrome103-ff102
```
Comment (devbugging) [inline .github/workflows/ci.yml]: Couldn't we use an existing cypress actions as seen here https://docs.cypress.io/guides/continuous-integration/github-actions#Testing-in-Chrome-and-Firefox-with-Cypress-Docker-Images 
?

**Review by devbugging — APPROVED** [review]

**Review by alse — APPROVED** [review]
Looks good!

**Pull request merged**